src/System.Management.Automation/engine/ArgumentToVersionTransformationAttribute.cs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Management.Automation
+{
+    /// <summary>
+    /// To make it easier to specify a version, we add some conversions that wouldn't happen otherwise:
+    ///   * A simple integer, i.e. 2;
+    ///   * A string without a dot, i.e. "2".
+    /// </summary>
+    internal class ArgumentToVersionTransformationAttribute : ArgumentTransformationAttribute
+    {
+        private static readonly IReadOnlyDictionary<string, Version> _empty = new Dictionary<string, Version>();
+
+        private readonly IReadOnlyDictionary<string, Version> _map;
```

Comment:
By looking at the code, the sole purpose of having this dictionary is for `latest` to work for `Set-StrictMode` command.
Let's not depending on this dictionary. Instead, add a new virtual method `virtual Version ConvertFromString(string )`, which simply returns `null` in this type. Then the transformation attribute for `Set-StrictMode` can handle `latest` in its override method.

When `ConvertFromString` returns `null`, proceed on with the other transformation steps, otherwise, return the result directly.